KlayGE/Core/Src/Base/HWDetect.cpp

```diff
@@ -108,9 +108,10 @@ namespace KlayGE
 		{
 			BOOST_ASSERT(wbem_services_);
 
-			wchar_t query_lang[] = L"WQL";
+			BSTR query_lang = SysAllocString(L"WQL");
```

Comment:
BSTR is wchar_t* on WIndows. No need to change this file.

Reply:
Unfortunately it is not correct. Read https://learn.microsoft.com/en-us/previous-versions/windows/desktop/automat/bstr:

A BSTR is a composite data type that consists of a length prefix, a data string, and a terminator. The following table describes these components.

The following code is incorrect:
 
> BSTR MyBstr = L"I am a happy BSTR";

This code builds (compiles and links) correctly, but it will not function properly because the string does not have a length prefix. If you use a debugger to examine the memory location of this variable, you will not see a four-byte length prefix preceding the data string.

Instead, use the following code:

> BSTR MyBstr = SysAllocString(L"I am a happy BSTR");

A debugger that examines the memory location of this variable will now reveal a length prefix containing the value 34. This is the expected value for a 17-byte character string that is represented as a wide-string literal through the inclusion of the "L" string prefix. The debugger will also show the terminator that appears after the data string.

**If you pass a wide-string literal as an argument to a COM function that is expecting a BSTR, the COM function behaves unexpectedly.**

Reply:
However, in many examples from Microsoft, wide strings are directly used as BSTR. E.g, https://learn.microsoft.com/en-us/windows/win32/api/wbemcli/nn-wbemcli-iwbemservices?source=recommendations, https://learn.microsoft.com/en-us/windows/win32/api/wbemcli/nf-wbemcli-iwbemlocator-connectserver. In the implementation, they actually accept strings with or without length prefix. It doesn't matter at least when using with IWbemServices.

Reply:
Oh, even their docs are inconsistent and contain errors. I would prefer to use recommended way and create MR to fix incorrect usage in docs you have mentioned.
Anyway, you decide which way should we use.

Reply:
I've landed 2 PRs to update Microsoft docs:
https://github.com/MicrosoftDocs/sdk-api/pull/1511
https://github.com/MicrosoftDocs/sdk-api/pull/1512

Now BSTR and wchar_t[] will not be intermixed and samples provide correct way of WMI API usage